craft_providers/multipass/multipass.py

```diff
@@ -386,3 +387,65 @@ def umount(self, *, mount: str) -> None:
                 brief=f"Failed to unmount {mount!r}.",
                 details=errors.details_from_called_process_error(error),
             ) from error
+
+    def wait_until_ready(
+        self, *, retry_wait: float = 0.25, timeout: Optional[float] = None
+    ):
+        """Wait until Multipass is ready (upon install/startup).
+
+        :param retry_wait: Time to sleep between retries.
+        :param timeout: Optional time.time() stamp to timeout after.
+        """
+        while timeout is None or time.time() < timeout:
+            multipass_version, multipassd_version = self.version()
+
+            if multipassd_version is not None:
+                return (multipass_version, multipassd_version)
+
+            time.sleep(retry_wait)
+
+        raise MultipassError(
+            brief="Timed out waiting for Multipass to become ready.",
+        )
+
+    def version(self):
+        """Get multipass and multipassd versions.
+
+        Split multipass version output should look like:
+
+            - ['multipass', '1.5.0']
+
+            - ['multipass', '1.5.0', 'multipassd', '1.5.0']
+
+            - ['multipass', '1.5.0+mac', 'multipassd', '1.5.0+mac']
+
+            - ['multipass', '1.5.0+win', 'multipassd', '1.5.0+win']
+
+        In some cases, Multipass may spew additional data, so protect against
+        it.  See: https://github.com/canonical/multipass/issues/2020
+
+        :returns: Tuple of parsed versions (multipass, multipassd).  multipassd
+                  may be None if Multipass is not yet ready.
```

Comment:
It could be useful to return some kind of mapping between component and version, such as a dictionary from component name to version, or a list of (component, version) tuples, e.g. to allow a more uniform return format for the `version()` call for different providers.

Reply:
I don't know if we'll get any other components in the future.  At the moment I don't think we'd care about them.  I'm returning it for completeness and because we use it to identify if multipass is ready.  As it is, the only upcoming usage of it in craft providers ignores the the second (multipassd) value.  How about we keep this for now and I'll revisit in the future when multipass supports a --format=yaml option here (open ticket).

Reply:
Also, fwiw this `version()` call is not really exposed to the applications btw.